core/dbt/deprecations.py

```diff
@@ -87,6 +87,18 @@ class AdapterDeprecationWarning(DBTDeprecation):
     deprecations[dep.name] = dep
 
 
+class MetricAttributesRenamed(DBTDeprecation):
+    _name = "metric-attr-renamed"
+    _description = """\
+dbt-core v1.3 renamed attributes for metrics:
+\n  'sql'              -> 'expression'
+\n  'type'             -> 'calculation_method'
+\n  'type: expression' -> 'calculation_method: derived'
+\nThe old metric parameter names will be fully deprecated in v1.4.
+\nPlease remove them from the metric definition of metric '{metric_name}'
+"""
```

Comment:
Issue created here: https://github.com/dbt-labs/dbt-core/issues/5849

Reply:
@emmyoop the link to this GH issue has been added to the deprecation warning!

Reply:
@callum-mcdata I actually meant a comment in the code, not as part of the deprecation message.  Sorry if that was unclear!  Unless you think users will find the link to the issue useful?

Reply:
🤦 woops! Although now that I think about it, if we include the link to the issue then anyone who feels very strongly that this behavior should be retained could go to it easily and let us know there. That way the information isn't spread over multiple channels..... so actually I do think it would be useful!